src/tokens/blockchain.service.ts

```diff
@@ -88,15 +104,60 @@ export class BlockchainConnectorService {
     });
   }
 
+  // Check if retry condition matches the err that's been hit
+  private matchesRetryCondition(err: any): boolean {
+    return this.retryCondition != '' && err?.toString().match(this.retryCondition) !== null;
+  }
+
+  // Delay by the appropriate amount of time given the iteration the caller is in
+  private async backoffDelay(iteration: number) {
+    const delay = Math.min(
+      this.retryBackOffInitial * Math.pow(this.retryBackOffFactor, iteration),
+      this.retryBackOffLimit,
+    );
+    await new Promise(resolve => setTimeout(resolve, delay));
+  }
+
+  // Generic helper function that makes an given blockchain function retryable
+  // by using synchronous, back-off delays for cases where the function returns
+  // an error which matches the configured retry condition
+  private async retryableCall<T = any>(
+    blockchainFunction: () => Promise<AxiosResponse<T>>,
+  ): Promise<AxiosResponse<T>> {
+    let response: any;
+    for (let retries = 0; retries <= this.retriesMax; retries++) {
+      try {
+        response = await blockchainFunction();
+        break;
+      } catch (e) {
+        if (this.matchesRetryCondition(e)) {
+          this.logger.debug(`Retry condition matched for error ${e}`);
+          // Wait for a backed-off delay before trying again
+          await this.backoffDelay(retries);
+        } else {
+          // Whatever the error was it's not one we will retry for
+          break;
+        }
+      }
+    }
+
+    return response;
```

Comment:
I think it's now possible for this to be undefined when it is returned - that doesn't seem to be correctly called out in the type specifiers or handled by the callers.